api/resolvers/item.js

```diff
@@ -635,18 +664,42 @@ export default {
         // count items: ((bid > ours.bid AND status = 'ACTIVE') OR (created_at > ours.created_at AND status <> 'STOPPED'))
         where = {
           OR: [
-            { maxBid: { gt: 0 }, status: 'ACTIVE' },
-            { createdAt: { gt: createdAt }, status: { not: 'STOPPED' } }
+            { boost: { gt: 0 } },
+            { createdAt: { gt: createdAt } }
           ]
         }
       }
 
-      where.subName = sub
+      where.AND = {
+        subName: sub,
+        status: 'ACTIVE',
+        deletedAt: null
+      }
       if (id) {
-        where.id = { not: Number(id) }
+        where.AND.id = { not: Number(id) }
       }
 
       return await models.item.count({ where }) + 1
+    },
+    boostPosition: async (parent, { id, sub, boost }, { models, me }) => {
+      if (boost <= 0) {
+        throw new GqlInputError('boost must be greater than 0')
+      }
+
+      const where = {
+        boost: { gte: boost },
+        status: 'ACTIVE',
+        deletedAt: null,
+        outlawed: false
+      }
+      if (id) {
+        where.id = { not: Number(id) }
+      }
+
+      return {
+        home: await models.item.count({ where }) === 0,
+        sub: await models.item.count({ where: { ...where, subName: sub } }) === 0
+      }
```

Comment:
See other comment about comment boosts showing `~undefined`:

If we fix that `sub` arrives as the string `"undefined"` here for comments using this:

```diff
diff --git a/components/adv-post-form.js b/components/adv-post-form.js
index 660160fe..43773fcb 100644
--- a/components/adv-post-form.js
+++ b/components/adv-post-form.js
@@ -90,8 +90,8 @@ export function BoostItemInput ({ item, sub, act = false, ...props }) {
   const [boost, setBoost] = useState(Number(item?.boost) + (act ? BOOST_MULT : 0))

   const [getBoostPosition, { data }] = useLazyQuery(gql`
-    query BoostPosition($id: ID, $boost: Int) {
-      boostPosition(sub: "${item?.subName || sub?.name}", id: $id, boost: $boost) {
+    query BoostPosition($sub: String, $id: ID, $boost: Int) {
+      boostPosition(sub: $sub, id: $id, boost: $boost) {
         home
         sub
       }
@@ -102,7 +102,7 @@ export function BoostItemInput ({ item, sub, act = false, ...props }) {

   useEffect(() => {
     if (boost) {
-      getPositionDebounce({ variables: { boost: Number(boost), id: item?.id } })
+      getPositionDebounce({ variables: { sub: item?.subName || sub?.name, boost: Number(boost), id: item?.id } })
     }
   }, [boost, item?.id])

```

we can always return `false` for `sub` for comments here like this:

```diff
diff --git a/api/resolvers/item.js b/api/resolvers/item.js
index fc8a1b03..c92001d9 100644
--- a/api/resolvers/item.js
+++ b/api/resolvers/item.js
@@ -698,7 +698,7 @@ export default {

       return {
         home: await models.item.count({ where }) === 0,
-        sub: await models.item.count({ where: { ...where, subName: sub } }) === 0
+        sub: sub ? await models.item.count({ where: { ...where, subName: sub } }) === 0 : false
       }
     }
   },
```

Additionally, I think we only want to run this over posts, not comments since they don't affect the ranking on the homepage or territories:

```diff
diff --git a/api/resolvers/item.js b/api/resolvers/item.js
index c92001d9..03bba2ca 100644
--- a/api/resolvers/item.js
+++ b/api/resolvers/item.js
@@ -690,7 +690,8 @@ export default {
         boost: { gte: boost },
         status: 'ACTIVE',
         deletedAt: null,
-        outlawed: false
+        outlawed: false,
+        parentId: null
       }
       if (id) {
         where.id = { not: Number(id) }
```